[RUMF-1262] add centralized documentation on page activity (#14097)
DataDog/documentation: content/en/real_user_monitoring/browser/monitoring_page_performance.md

Datadog provides a unique performance metric, `loading_time`, which calculates the time needed for a page to load. This metric works for both `initial_load` and `route_change` navigation.

### How loading time is calculated

To account for modern web applications, loading time watches for network requests and DOM mutations.

- **Initial Load**: Loading Time is equal to _whichever is longer_:

  - The difference between `navigationStart` and `loadEventEnd`.
  - Or the difference between `navigationStart` and the first time the page has no activity. Read [How page activity is calculated](#how-page-activity-is-calculated) for details.

- **SPA Route Change**: Loading Time is equal to the difference between the user click and the first time the page has no activity. Read [How page activity is calculated](#how-page-activity-is-calculated) for details.

### How page activity is calculated

> Whenever a navigation or a click occurs, the RUM SDK tracks the page activity to estimate the time until the interface is stable again. The page is deemed to have activity by looking at network requests and DOM mutations. The page activity ends when there are no ongoing requests and no DOM mutation for more than 100ms. The page is determined to have no activity if no requests or DOM mutation occurred in 100ms.

Reviewer: ```suggestion
Whenever a navigation or a click occurs, the Browser RUM SDK tracks the page activity to estimate the time until the interface is stable again. The page is deemed to have activity by looking at network requests and DOM mutations. The page activity ends when there are no ongoing requests and no DOM mutation for more than 100ms. The page is determined to have no activity if no requests or DOM mutation occurred in 100ms.
```

The criteria of 100ms since last request or DOM mutation might not be an accurate determination of activity in the following scenarios:

- The application collects analytics by sending requests to an API periodically or after every click.

- The application uses "[comet][16])" techniques (that is, streaming or long polling), and the request stays on hold for an indefinite time.

> To improve the accuracy of activity determination in these cases, specify `excludedActivityUrls`, a list of resources for the RUM SDK to exclude when computing the page activity:

Reviewer: ```suggestion
To improve the accuracy of activity determination in these cases, specify `excludedActivityUrls`, a list of resources for the Browser RUM SDK to exclude when computing the page activity:
```

- **Initial Load**: Loading Time is equal to _whichever is longer_:

  - The difference between `navigationStart` and `loadEventEnd`.

>   - Or the difference between `navigationStart` and the first time the page has no activity (see [next section][18] for details).

Reviewer 2: ```suggestion
  - Or the difference between `navigationStart` and the first time the page has no activity. Read [How page activity is calculated](#how-page-activity-is-calculated) for details.
```

> - **SPA Route Change**: Loading Time is equal to the difference between the user click and the first time the page has no activity (see [next section][18] for details).

Reviewer 2: ```suggestion
- **SPA Route Change**: Loading Time is equal to the difference between the user click and the first time the page has no activity. Read [How page activity is calculated](#how-page-activity-is-calculated) for details.
```

> ### How is page activity calculated?

Reviewer 2: ```suggestion
### How page activity is calculated
```

> Whenever a navigation or a click occurs, we start tracking the page activity to estimate the time until the interface is stable again. We consider that a page has activity by looking at network requests and DOM mutations. The page activity stops when there is no ongoing requests and no DOM mutation for more than 100ms. We consider that there was no activity if no requests or DOM mutation occurred in 100ms.

Reviewer 2: ```suggestion
Whenever a navigation or a click occurs, the RUM SDK tracks the page activity to estimate the time until the interface is stable again. The page is deemed to have activity by looking at network requests and DOM mutations. The page activity ends when there are no ongoing requests and no DOM mutation for more than 100ms. The page is determined to have no activity if no requests or DOM mutation occurred in 100ms.
```

> This approach works fine most of the time, but sometimes requests made by the application do not reflect actual activity as they don't have impact on the interface. In particular, we identified two problematic use cases:

Reviewer 2: The criteria of 100ms since last request or DOM mutation might not be an accurate determination of activity in the following scenarios:

> This approach works fine most of the time, but sometimes requests made by the application do not reflect actual activity as they don't have impact on the interface. In particular, we identified two problematic use cases:

Reviewer 2: ```suggestion
The criteria of 100ms since last request or DOM mutation might not be an accurate determination of activity in the following scenarios:
```

> - When the application is collecting analytics by sending requests to an API periodically or following every click.

Reviewer 2: ```suggestion
- The application collects analytics by sending requests to an API periodically or after every click.
```

> - When the application is using "[comet][17]" techniques (that is, streaming or long polling), the request stays on hold for an indefinite time.

Reviewer 2: ```suggestion
- The application uses "[comet][17]" techniques (that is, streaming or long polling), and the request stays on hold for an indefinite time.
```

> Such request have a negative impact on the page activity computed by RUM. To improve its accuracy, the RUM Browser SDK allows to provide a list of URLs of resources to exclude when computing the page activity. Example:

Reviewer 2: ```suggestion
To improve the accuracy of activity determination in these cases, specify `excludedActivityUrls`, a list of resources for the RUM SDK to exclude when computing the page activity:
```

});
```

> Once the timing is sent, the timing will be accessible as `@view.custom_timings.<timing_name>` (For example, `@view.custom_timings.first_scroll`). You must [create a measure][17] before graphing it in RUM analytics or in dashboards.

Reviewer: ```suggestion
Once the timing is sent, the timing is accessible as `@view.custom_timings.<timing_name>`, for example: `@view.custom_timings.first_scroll`. You must [create a measure][17] before graphing it in RUM Analytics or in dashboards.
```

[14]: https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event
[15]: https://developer.mozilla.org/en-US/docs/Web/API/History
[16]: /real_user_monitoring/explorer/?tab=measures#setup-facets-and-measures
[17]: https://en.wikipedia.org/wiki/Comet_(programming)

> [18]: #how-is-page-activity-calculated

Reviewer 2: ```suggestion
```
